src/rust/ensogl/lib/core/src/animation/easing.rs

```diff
@@ -323,3 +331,70 @@ impl<T:Value,F,Cb> Animator<T,F,Cb> where F:AnyFnEasing, Cb:Callback<T> {
         self.data.duration.set(t);
     }
 }
+
+
+// =====================
+// === AnimationLoop ===
+// =====================
+
+/// A wrapper over animation loop implementation. This type is defined mainly to make Rust type
+/// inferencer happy (not infer infinite, recursive types).
+#[derive(CloneRef,Derivative)]
+#[derivative(Clone(bound=""))]
+#[derivative(Default(bound=""))]
+#[allow(clippy::type_complexity)]
+#[allow(missing_debug_implementations)]
+pub struct AnimationLoop<T,F,OnStep,OnEnd> {
+    animation_loop : Rc<CloneCell<Option<AnimationStep<T,F,OnStep,OnEnd>>>>,
```

Comment:
Should there be a type alias for this? That would also avoid the Clippy warning here (and in other places further down).

Reply:
Well, I think the alias would have sense if its name actually give some additional information. But I can't think out any other name than "OptAnimationStep".

Reply:
This type is not used anywhere else and its just a hack to hide the type from inferencer in order to not make it loop itself, so I dont think alias would make the code more clear, but if you really want it, I can do it.